src/os/bluestore/bluefs_types.cc

```diff
@@ -200,17 +200,23 @@ void bluefs_fnode_t::generate_test_instances(list<bluefs_fnode_t*>& ls)
   ls.back()->size = 1048576;
   ls.back()->mtime = utime_t(123,45);
   ls.back()->extents.push_back(bluefs_extent_t(0, 1048576, 4096));
-  ls.back()->__unused__ = 1;
+  ls.back()->type = 0;
 }
 
 ostream& operator<<(ostream& out, const bluefs_fnode_t& file)
 {
+  static constexpr std::string_view node_types[] =  {
+    "LEGACY",
+    "WAL_V2",
+  };
+
   return out << "file(ino " << file.ino
 	     << " size 0x" << std::hex << file.size << std::dec
 	     << " mtime " << file.mtime
 	     << " allocated " << std::hex << file.allocated << std::dec
 	     << " alloc_commit " << std::hex << file.allocated_commited << std::dec
 	     << " extents " << file.extents
+	     << " type " << node_types[file.type] << std::dec
```

Comment:
The printing is weird now.
`
2024-05-10T12:11:30.862+0000 7fe5bea18b40 10 bluefs open_for_read db/CURRENT (sequential)
2024-05-10T12:11:30.862+0000 7fe5bea18b40 10 bluefs open_for_read h 0x55752ad246c0 on file(ino 33 size 0x10 mtime 2024-05-10T11:54:22.519090+0000 allocated 10000 alloc_commit 10000 extents [1:0x32000~10000] type LEGACY)
`
I think the string "wal_v2" should only be printed for wal_v2 files, otherwise nothing.

Reply:
I like explicitly stating a file is now "legacy" or not but I'll change it if you prefer it.